### ISSUE TITLE
Extend String class with Inflector methods

### ISSUE THREAD

**Issue by actsasflinn** [luckyframework/inflector#7]

**Comment by paulcsmith** [comment]
I’m not totally sure about extending string classes. Or maybe it can be optional with the default being not to extend built in classes. 

The reason is that it can mess with other libraries or if standard lib adds the same method with a slightly different implementation it could also less things up. I think extending classes can sometimes be ok, but for something like this I think calling the inflection methods or including the inflector module in a class is probably ok.

**Comment by paulcsmith** [comment]
What do you think?

**Comment by actsasflinn** [comment]
Not a bad idea to make it optional and I do think the inflector has some value stand alone, perhaps even outside of Lucky. I'm not big on separating the extensions out completely since one would never use it without the Inflector. If the Inflector and String extensions are in the same place, it might be easier to keep them in sync is my guess?

**Comment by actsasflinn** [comment]
I kind of think LuckyInflector::Inflector.pluarlize(word) is pretty verbose too.

**Comment by paulcsmith** [comment]
@actsasflinn You're right. Separating to another repo is unnecessary and would probably lead to complications. What we could do though is not require the core extensions by default, but you could require them with `require "lucky_inflector/core_extensions/**" Similar to how LuckyWeb handles and requires tasks: https://github.com/luckyframework/cli/blob/master/src/web_app_skeleton/tasks.cr#L11 and https://github.com/luckyframework/web/tree/master/tasks

That way people can include it if they want, but they don't have to. I think all of the Lucky packages should not use core extensions. That way people aren't required to extend core classes/structs to use Lucky.

If you'd like to start on that I would love it! I think this is a good combo of allowing people to do what they want, but not forcing them too. Thoughts?

**Comment by actsasflinn** [comment]
Yup, I like it, I'll adjust this a bit. I'm working on this as a part of the LuckyWeb::TextHelpers work, so I'll make them use the package instead of String extensions also.

**Comment by actsasflinn** [comment]
I’ll work on this in a few days, I’m without my laptop.
